gr-audio/lib/alsa/alsa_sink.cc

```diff
@@ -369,25 +369,42 @@ int alsa_sink::work_s32(int noutput_items,
     unsigned int nchan = input_items.size();
     const float** in = (const float**)&input_items[0];
     sample_t* buf = (sample_t*)d_buffer;
-    int bi;
+
+    /*
+     * vector of vectors to keep each channel after conversion
+     * to int32_t. These are then placed in the buffer in the
+     * standard WAV file format:
+     * [chan0samp0, chan1samp0, ..., chanNsamp0, chan0samp1, ...]
+     * */
+    volk::vector<volk::vector<sample_t>> buf_channels(
+        nchan, volk::vector<sample_t>(d_period_size));
+
     int n;
 
     unsigned int sizeof_frame = nchan * sizeof(sample_t);
     assert(d_buffer_size_bytes == d_period_size * sizeof_frame);
 
     for (n = 0; n < noutput_items; n += d_period_size) {
         // process one period of data
-        bi = 0;
-        for (unsigned int i = 0; i < d_period_size; i++) {
-            for (unsigned int chan = 0; chan < nchan; chan++) {
-                buf[bi++] = (sample_t)(in[chan][i] * scale_factor);
-            }
-        }
+        for (unsigned int chan = 0; chan < nchan; chan++)
+            volk_32f_s32f_convert_32i(
```

Comment:
You should include `volk.h` in this .cc: "Include what you use" is a very desirable pattern.

Reply:
Sounds good. Thanks!